libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -157,27 +187,75 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
   Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<FeatureFlagValue>[] {
-    const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+  ): SetConfigurationSettingParam<MsFeatureFlagValue>[] {
+    const settings = new Array<SetConfigurationSettingParam<MsFeatureFlagValue>>();
+    const featureFlags = new Array<MsFeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
+    if (this.dotnetFmSchemaKeyWord) {
+      const dotnetSchemaFeatureFlags = this.getDotnetFmSchemaFeatureFlags(config);
+
+      const featureManagementIndex: number = Constants.FeatureManagementKeyWords.indexOf(this.dotnetFmSchemaKeyWord);
```

Comment:
Line 197 can be declared next to where its being used

Reply:
Its being referenced in the for loop but I don't want to add it within the for loop since we only need to evaluate it once

Reply:
True I hadn't seen that line 206 - 210 is within a for loop